File: examples/undocumented/python_modular/classifier_lda_modular.py
#!/usr/bin/env python
traindat = '../data/fm_train_real.dat'
testdat = '../data/fm_test_real.dat'
label_traindat = '../data/label_train_twoclass.dat'

parameter_list = [[traindat,testdat,label_traindat,3,1],[traindat,testdat,label_traindat,4,1]]

def classifier_lda_modular (train_fname=traindat,test_fname=testdat,label_fname=label_traindat,gamma=3,num_threads=1):
	from modshogun import RealFeatures, BinaryLabels, LDA, CSVFile

	feats_train=RealFeatures(CSVFile(train_fname))
	feats_test=RealFeatures(CSVFile(test_fname))
	labels=BinaryLabels(CSVFile(label_fname))

	lda=LDA(gamma, feats_train, labels)
	lda.train()

	bias=lda.get_bias()
	w=lda.get_w()
	predictions = lda.apply(feats_test).get_labels()
	return lda,predictions

if __name__=='__main__':
	print('LDA')
	classifier_lda_modular(*parameter_list[0])

File: examples/undocumented/r_modular/classifier_lda_modular.R
library(shogun)

fm_train_real <- t(as.matrix(read.table('../data/fm_train_real.dat')))
fm_test_real <- t(as.matrix(read.table('../data/fm_test_real.dat')))
label_train_twoclass <- as.double(read.table('../data/label_train_twoclass.dat')$V1)

# lda
print('LDA')

feats_train <- RealFeatures()
dump <- feats_train$set_feature_matrix(fm_train_real)
feats_test <- RealFeatures()
dump <- feats_test$set_feature_matrix(fm_test_real)

gamma <- 3
labels <- BinaryLabels()
labels$set_labels(label_train_twoclass)

lda <- LDA(gamma, feats_train, labels)
dump <- lda$train()

dump <- lda$get_bias()
dump <- lda$get_w()
dump <- lda$set_features(feats_test)
lab <- lda$apply()
out <- lab$get_labels()
